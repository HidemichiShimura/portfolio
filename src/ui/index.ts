// Atoms
export { default as PageHeading } from "./atoms/PageHeading/PageHeading";
export { default as Avatar } from "./atoms/Avatar/Avatar";
export { default as Typography } from "./atoms/Typography/Typography";
export { default as SectionHeading } from "./atoms/SectionHeading/SectionHeading";
export { default as Icon } from "./atoms/Icon/Icon";
export { default as RadialLinearContainer } from "./atoms/RadialLinearContainer/RadialLinearContainer";
export { default as Grid } from "./atoms/Grid/Grid";
export { default as GradientCard } from "./atoms/GradientCard/GradientCard";
export { default as Card } from "./atoms/Card/Card";

// Molecules
export { default as HeroHeadings } from "./molecules/HeroHeadings/HeroHeadings";
