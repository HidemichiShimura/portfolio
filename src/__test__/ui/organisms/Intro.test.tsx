import { Intro } from "@/ui";
import { screen, render } from "@testing-library/react";

const TEXT =
  "I am a self-taught web developer with a strong background in modern frontend and backend skills with experiences as a system admin for servers and networks. I have a strong passion for problem-solving and am always eager to assist my team members. I thrive in fast-paced environments with new challenges. Mutual assistance is my top priority.";

describe("Organisms: <Intro />", () => {
  test("renders", () => {
    render(<Intro />);

    expect(screen.queryByTitle("Intro")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    render(<Intro />);

    expect(screen.queryByTitle("Intro")).toMatchSnapshot();
  });

  test("displays my description text", () => {
    render(<Intro />);

    expect(screen.queryByText(TEXT)).toBeInTheDocument();
  });
});
