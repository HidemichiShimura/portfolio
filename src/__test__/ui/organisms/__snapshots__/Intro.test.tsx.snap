// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: <Intro /> matches snapshot 1`] = `
<section
  class="flex h-screen items-center md:h-96"
  title="Intro"
>
  <p
    class="text-light-gray text-base font-sans"
  >
    I am a self-taught web developer with a strong background in modern frontend and backend skills with experiences as a system admin for servers and networks. I have a strong passion for problem-solving and am always eager to assist my team members. I thrive in fast-paced environments with new challenges. Mutual assistance is my top priority.
  </p>
</section>
`;
